projects/storefrontlib/src/cms-components/checkout/guards/payment-details-set.guard.ts

```diff
@@ -29,23 +30,31 @@ export class PaymentDetailsSetGuard implements CanActivate {
       );
     }
 
-    if (checkoutStep && checkoutStep.disabled) {
-      return of(true);
-    }
-
-    return this.checkoutDetailsService
-      .getPaymentDetails()
-      .pipe(
-        map((paymentDetails) =>
-          paymentDetails && Object.keys(paymentDetails).length !== 0
+    return combineLatest([
+      this.paymentTypeService.getSelectedPaymentType(),
+      this.checkoutDetailsService.getPaymentDetails(),
+    ]).pipe(
+      filter(([selected, _]) => selected !== undefined),
```

Comment:
When destructuring an array, I think you can simply omit the values you don't want when starting from the beginning.  Said differently, you could use `[selected]` instead of `[selected, _]` if you don't need the `_`.  Your call.

Reply:
done